src/figuretype/supplier.c

```diff
@@ -168,6 +168,27 @@ static int change_market_supplier_destination(figure *f, int dst_building_id)
     return 1;
 }
 
+static int new_market_dest_better_than_old_market_dest(figure *f, resource_type r, resource_storage_info info[RESOURCE_MAX])
```

Comment:
```suggestion
static int is_better_destination(figure *f, resource_type r, resource_storage_info *info)
```

You only need to return the pointer to the information for the resource `r`, not the whole array.